test/integration/generated/datasearch/src/models/parameters.ts

```diff
@@ -54,7 +54,7 @@ export const indexName: OperationURLParameter = {
 };
 
 export const xMsClientRequestId: OperationParameter = {
-  parameterPath: ["options", "requestOptionsParam", "xMsClientRequestId"],
+  parameterPath: ["options", "requestOptions", "xMsClientRequestId"],
```

Comment:
looks like this one still has a definition of requestOptionsParam. Here's how this parameter being called in the operation.
```
  /**
   * Queries the number of documents in the index.
   * @param options The options parameters.
   */
  count(
    options?: DocumentsCountOptionalParams
  ): Promise<DocumentsCountResponse> {
    return this.client.sendOperationRequest({ options }, countOperationSpec);
  }
```
Here's how the model being defined in models/index.ts
```
export interface DocumentsCountOptionalParams
  extends coreClient.OperationOptions {
  /** Parameter group */
  requestOptionsParam?: RequestOptions;
}
export interface RequestOptions {
  /** The tracking ID sent with the request to help with debugging. */
  xMsClientRequestId?: string;
}
```
@joheredi maybe we should keep requestOptions in this case ?

Reply:
Yeah parameter path should match the structure of the generated object. For example here, accessing `xMsClientRequestId`, which is a property of `requestOptionsParam` in the interface  `DocumentsCountOptionalParams`. And contained inside the `options` parameter in the `count` operation in `documents.ts`. The parameter path should look like if you were accessing that property i.e. `options.requestOptionsParam.xMsClientRequestId` so in this case, the right parameter path would be `["options", "requestOptionsParam", "xMsClientRequestId"]`

So the new parameter path `requestOptions` would be incorrect. You'd either need to not change it or also change the interface  `DocumentsCountOptionalParams` to have `requestOptions` instead of `requestOptionsParam`

Reply:
changed the parameterPath back to requestOptionsParam